src/cli/rustup_mode.rs

```diff
@@ -911,31 +923,16 @@ fn check_updates(cfg: &Cfg) -> Result<utils::ExitCode> {
         }
     }
 
-    // Get current rustup version
-    let current_version = env!("CARGO_PKG_VERSION");
-
-    // Get available rustup version
-    let available_version = get_available_rustup_version()?;
-
-    let _ = t.attr(term2::Attr::Bold);
-    write!(t, "rustup - ")?;
+    check_rustup_update()?;
 
-    if current_version != available_version {
-        let _ = t.fg(term2::color::YELLOW);
-        write!(t, "Update available")?;
-        let _ = t.reset();
-        writeln!(t, " : {} -> {}", current_version, available_version)?;
-    } else {
-        let _ = t.fg(term2::color::GREEN);
-        write!(t, "Up to date")?;
-        let _ = t.reset();
-        writeln!(t, " : {}", current_version)?;
-    }
     Ok(utils::ExitCode(0))
 }
 
 fn update(cfg: &mut Cfg, m: &ArgMatches<'_>) -> Result<utils::ExitCode> {
-    let self_update = !m.is_present("no-self-update") && !self_update::NEVER_SELF_UPDATE;
+    let self_update_mode = cfg.get_self_update_mode()?;
+    let self_update = !m.is_present("no-self-update")
+        && !self_update::NEVER_SELF_UPDATE
+        && self_update_mode == "enable";
```

Comment:
I am not sure I am interpreting this boolean expression correctly - and even if I am, I'm not certain of the intent or user story here; this new feature needs documenting, and that would be a good place to describe the interactions of all the various bits:
- it might be compiled out
- it might be controlled via the command line
- it might be controlled via state

Do places of settings have precedence, or values of settings, or some mixture of both? Lets write it down clearly and see if it makes sense.

Reply:
Added. Update only if rustup does **not** have the no-self-update feature, and auto-self-update is configured to **enable** and has **no** no-self-update parameter.

All other cases will not be updated.